src/compiler/factory/nodeFactory.ts

```diff
@@ -6829,8 +6847,8 @@ namespace ts {
             node.declarationText = declarationTextOrJavascriptPath;
             node.declarationMapPath = declarationMapPath;
             node.declarationMapText = declarationMapTextOrBuildInfoPath;
-            node.javascriptPath = javascriptPath;
-            node.declarationPath = declarationPath;
+            node.javascriptPath = javascriptPathOrHost as string | undefined;
+            node.declarationPath = declarationPathOrOptions as string | undefined;

```

Comment:
Do the declaration of these types need to change? Seems like these assume string already but this is just casting host to a string.

Reply:
This is just wiring the overload methods. (just like declarationMapTextOrBuildInfoPath)

Reply:
Right, but `declarationMapTextOrBuildInfoPath` is definitely a `string | undefined`, so it makes sense.

I guess I'm asking why the `InputFiles` isn't being modified and we're casting `CompilerHost` or `CompilerOptions` to string.

Reply:
It works out because of the way overloads are written but let me see if i can simplify because this has always been tricky with public and internal APIs and the names are confusing esp everything being string etc